src/write/macho.rs

```diff
@@ -546,7 +546,7 @@ impl<'a> Object<'a> {
                     let r_length = match reloc.size {
                         8 => 0,
                         16 => 1,
-                        32 => 2,
+                        12 | 21 | 26 | 32 => 2,
```

Comment:
Maybe instead of this, change `reloc.size` to 32 in `macho_fixup_relocation` while you're doing the other special handling.